src/Components/WorkloadRules/WorkloadsRules.test.js

```diff
@@ -75,3 +77,73 @@ describe('pruneWorkloadsRulesFilters function', () => {
     ]);
   });
 });
+
+// Mocking SortByDirection for testing purposes
+const SortByDirection = {
+  asc: 'asc',
+  desc: 'desc',
+};
+
+describe('switchSort function', () => {
+  test('should return details for sortIndex 1', () => {
+    const item = [{ rule: { details: 'someDetails' } }];
+    const result = switchSort(1, item);
+    expect(result).toBe('someDetails');
+  });
+
+  test('should return total_risk for sortIndex 2', () => {
+    const item = [{ rule: { total_risk: 42 } }];
+    const result = switchSort(2, item);
+    expect(result).toBe(42);
+  });
+
+  test('should return objects length for sortIndex 3', () => {
+    const item = [{ rule: { objects: [1, 2, 3] } }];
+    const result = switchSort(3, item);
+    expect(result).toBe(3);
+  });
+
+  test('should return modified for sortIndex 4', () => {
+    const item = [{ rule: { modified: '2024-01-10' } }];
+    const result = switchSort(4, item);
+    expect(result).toBe('2024-01-10');
+  });
+
+  test('should return 0 for invalid sortIndex', () => {
+    const item = [{ rule: {} }];
+    const result = switchSort(5, item);
+    expect(result).toBe(0);
+  });
+});
+
+describe('sortWithSwitch function', () => {
+  test('should return original rows if sortIndex is invalid or firstRule is true', () => {
+    const filteredRows = [{ rule: { details: 'abc' } }];
+    const result = sortWithSwitch(5, SortByDirection.asc, filteredRows, false);
+    expect(result).toEqual(filteredRows);
+  });
+
+  test('should sort rows in ascending order based on switchSort', () => {
+    const filteredRows = [
+      { rule: { details: 'abc' } },
+      { rule: { details: 'xyz' } },
+    ];
+    const result = sortWithSwitch(1, SortByDirection.asc, filteredRows, false);
+    expect(result).toEqual([
+      { rule: { details: 'abc' } },
+      { rule: { details: 'xyz' } },
+    ]);
+  });
+
+  test('should sort rows in descending order based on switchSort', () => {
+    const filteredRows = [
+      { rule: { details: 'abc' } },
+      { rule: { details: 'xyz' } },
+    ];
+    const result = sortWithSwitch(1, SortByDirection.desc, filteredRows, false);
+    expect(result).toEqual([
+      { rule: { details: 'abc' } },
+      { rule: { details: 'xyz' } },
```

Comment:
Shouldn't this be an opposite? If we sort in the **descending** order...

Reply:
Yep, I updated the tests